earth/earth_test.go

```diff
@@ -0,0 +1,193 @@
+		{s2.AvgAreaMetric.Value(20), 77.31706517582228 * unit.SquareMeter},      // average area of level 20 cells
+		{s2.AvgAreaMetric.Value(30), 73.73529927808979 * unit.SquareMillimeter}, // average area of level 30 cells
+		{s2.PolygonFromLoops([]*s2.Loop{s2.EmptyLoop()}).Area(), 0 * unit.SquareMeter},
+	}
+	for _, test := range tests {
+		if got, want := AreaFromSteradians(test.steradians), test.area; !float64Eq(got.SquareMeters(), want.SquareMeters()) {
+			t.Errorf("AreaFromSteradians(%v) = %v, want %v", test.steradians, got, want)
+		}
+	}
+}
+
+func TestSteradiansFromArea(t *testing.T) {
+	const earthArea = 6371.01 * 6371.01 * math.Pi * 4 * unit.SquareKilometer
+	tests := []struct {
+		steradians float64
+		area       unit.Area
+	}{
+		{1, earthArea / 4 / math.Pi},
+		{s2.PolygonFromLoops([]*s2.Loop{s2.FullLoop()}).Area(), earthArea},
+		{s2.PolygonFromLoops([]*s2.Loop{s2.LoopFromPoints([]s2.Point{
+			s2.PointFromLatLng(s2.LatLngFromDegrees(-90, 0)),
+			s2.PointFromLatLng(s2.LatLngFromDegrees(0, 0)),
+			s2.PointFromLatLng(s2.LatLngFromDegrees(90, 0)),
+			s2.PointFromLatLng(s2.LatLngFromDegrees(0, -90)),
+		})}).Area(), earthArea / 4},
+		{s2.CellFromCellID(s2.CellIDFromFace(2)).ExactArea(), earthArea / 6},
+		{s2.AvgAreaMetric.Value(10), 81.07281893380302 * unit.SquareKilometer},  // average area of level 10 cells
+		{s2.AvgAreaMetric.Value(20), 77.31706517582228 * unit.SquareMeter},      // average area of level 20 cells
+		{s2.AvgAreaMetric.Value(30), 73.73529927808979 * unit.SquareMillimeter}, // average area of level 30 cells
+		{s2.PolygonFromLoops([]*s2.Loop{s2.EmptyLoop()}).Area(), 0 * unit.SquareMeter},
+	}
+	for _, test := range tests {
+		if got, want := SteradiansFromArea(test.area), test.steradians; !float64Eq(got, want) {
+			t.Errorf("SteradiansFromArea(%v) = %v, want %v", test.area, got, want)
+		}
+	}
+}
+
+func TestInitialBearingFromLatLngs(t *testing.T) {
+	for _, tc := range []struct {
+		name string
+		a, b s2.LatLng
+		want s1.Angle
+	}{
+		{"Westward on equator", s2.LatLngFromDegrees(0, 50),
+			s2.LatLngFromDegrees(0, 100), s1.Degree * 90},
+		{"Eastward on equator", s2.LatLngFromDegrees(0, 50),
+			s2.LatLngFromDegrees(0, 0), s1.Degree * -90},
+		{"Northward on meridian", s2.LatLngFromDegrees(16, 28),
+			s2.LatLngFromDegrees(81, 28), s1.Degree * 0},
+		{"Southward on meridian", s2.LatLngFromDegrees(24, 64),
+			s2.LatLngFromDegrees(-27, 64), s1.Degree * 180},
+		{"Towards north pole", s2.LatLngFromDegrees(12, 76),
+			s2.LatLngFromDegrees(90, 50), s1.Degree * 0},
+		{"Towards south pole", s2.LatLngFromDegrees(-35, 105),
+			s2.LatLngFromDegrees(-90, -120), s1.Degree * 180},
+		{"Spain to Japan", s2.LatLngFromDegrees(40.4379332, -3.749576),
+			s2.LatLngFromDegrees(35.6733227, 139.6403486), s1.Degree * 29.2},
+		{"Japan to Spain", s2.LatLngFromDegrees(35.6733227, 139.6403486),
+			s2.LatLngFromDegrees(40.4379332, -3.749576), s1.Degree * -27.2},
+	} {
```

Comment:
// If a == b, a == -b, or a is one of the Earth's poles, the return value is undefined.

Will it crash or just give an indeterminate answer if you added one or two of these cases?

Reply:
Added a test to ensure it returns a real, but unspecified value.  Test comment explains why the value is undefined.